util/conn/manager.go

```diff
@@ -0,0 +1,105 @@
+package conn
```

Comment:
This package needs a README and doc comments, but I'll add that myself afterwards.